### PR TITLE
1835 use engine

### PR DESCRIPTION
**Pull request by IagoLast**
https://github.com/CartoDB/cartodb.js/issues/1835
---

### CODE REVIEW

File: src/geo/map/legends/legend-model-base.js [hunk 1]
```diff
@@ -15,13 +16,11 @@ var LegendModelBase = Backbone.Model.extend({
...
-    deps.visModel.on('reload', this._onVisReloading, this);
-  },
-
-  _onVisReloading: function () {
-    this.set('state', this.constructor.STATE_LOADING);
+    if (!deps.engine) throw new Error('engine is required');
+    var self = this;
+    deps.engine.on(Engine.Events.RELOAD_STARTED, function () {
+      self.set('state', self.constructor.STATE_LOADING);
+    });
```
Comment (Jesus89) [inline src/geo/map/legends/legend-model-base.js]: Use .bind(this)

File: src/engine.js [hunk 2]
```diff
@@ -97,15 +98,20 @@ Engine.prototype.getLayerGroup = function () {
...
  * @api
  */
-Engine.prototype.on = function on (event, callback) {
-  this._eventEmmitter.on(event, callback);
+Engine.prototype.on = function on (event, callback, context) {
+  if (context) {
+    this._eventEmmitter.on(event, callback, context);
+  } else {
+    this._eventEmmitter.on(event, callback);
+  }
```
Comment (Jesus89) [inline src/engine.js]: Use this._eventEmmitter.on(event, callback, context);

File: src/engine.js [hunk 3]
```diff
@@ -114,8 +120,12 @@ Engine.prototype.on = function on (event, callback) {
...
  * @api
  */
-Engine.prototype.off = function off (event, callback) {
-  this._eventEmmitter.off(event, callback);
+Engine.prototype.off = function off (event, callback, context) {
+  if (context) {
+    this._eventEmmitter.off(event, callback, context);
+  } else {
+    this._eventEmmitter.off(event, callback, context);
+  }
```
Comment (Jesus89) [inline src/engine.js]: Use this._eventEmmitter.on(event, callback, context);

**Comment by IagoLast** [comment]
LGTM

**Pull request merged**